### PR TITLE
Fix empty workspace name error in FDA transform tab

### LINKED ISSUE #29212: FDA Error
Open FDA
Load HIFI 84447
Go to transform and do an FFT

Change the real workspace
Change the imaginary workspace
Click calculate FFT - exception

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
Due to recent changes in the FDA interface the transform tab needs to contain workspaces which have been selected in the grouping tab (as this is what the fitting tab expects). This PR makes it so the grouping tab will inform and update the transform tab when groups/pairs are selected.

**To test:**

- Open FDA
- Load HIFI 84447
- Go to transform tab
- You should be able to perform a transform on each combination, with the plot updating for each one
- To update workspace choices you'll have to add groups to the analysis (in the grouping tab).

Fixes #29212

This does not require release notes as the issue was introduced in this sprint.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/fft_presenter_context_interaction_test.py [hunk 1]
```diff
@@ -301,25 +312,39 @@ def test_calculate_FFT_calls_correct_algorithm_sequence_with_imaginary(self, fie
...
+            'Transform': 'Forward',
+            'AcceptXRoundingErrors': True,
+            'AutoShift': True,
+            'InputImagWorkspace': apodization_mock_return,
+            "Imaginary": 0})
         field_mock.assert_called_once_with(fft_mock_return)
-        self.presenter.add_fft_workspace_to_ADS.assert_called_once_with(name,Im_name,field_mock_return)
+        self.presenter.add_fft_workspace_to_ADS.assert_called_once_with(name, Im_name, field_mock_return)
+
+    def test_selection_removed_genWorkspaceName(self):
```
Comment (Harrietbrown) [inline scripts/test/Muon/fft_presenter_context_interaction_test.py:334]: Can you also add a test for the addition of a new group updating the list?

**Review by Harrietbrown — DISMISSED** [review]
The list updates as it should. Could you just add one other unit test?

**Review by Harrietbrown — COMMENTED** [review]
The test looks good. I have noticed that the groups are added to the list in the order they were selected instead of the order they are listed in the table on the grouping tab. This is a minor issue so if it would be more effor than it is worth to fix I would be ok with going forward as is. What do you think?

**Comment by StephenSmith25** [comment]
Yeah unfortunately I think that’s just the way they are added to the context when they are selected.

**Review by Matt-Cumber — APPROVED** [review]
Nice clean up of the code. The latest tests look good and is clear. 

Just noticed should you connect this with an issue on ZenHub?

**Pull request merged**